### PR TITLE
Archive the gpdb src codes with submodules

### PR DESCRIPTION
**Pull request by baotingfang**
[#168285479]

Authored-by: Tingfang Bao <bbao@pivotal.io>

### CODE REVIEW

File: .yamllint [hunk 1]
```diff
@@ -17,7 +17,7 @@ rules:
   document-start:
     level: warning
   empty-lines: enable
-  empty-values: enable
+  empty-values: disable
```
Comment (amilkh) [inline .yamllint]: I think this should be its own separate commit? commit message could be like "Disable empty-values for yamllint\nMany of our concourse task .yml files have empty value variables
https://yamllint.readthedocs.io/en/stable/rules.html#module-yamllint.rules.empty_values

File: ci/concourse/scripts/gpdb_github_release.bash [hunk 2]
```diff
@@ -13,20 +13,44 @@
...
 	pushd "${BASE_DIR}/gpdb_src"
 	git --no-pager show --summary refs/tags/"${GPDB_RELEASE_TAG}"
 
-	git archive -o "${BASE_DIR}/release_artifacts/${GPDB_RELEASE_TAG}.tar.gz" --prefix="gpdb-${GPDB_RELEASE_TAG}/" --format=tar.gz refs/tags/"${GPDB_RELEASE_TAG}"
-	git archive -o "${BASE_DIR}/release_artifacts/${GPDB_RELEASE_TAG}.zip" --prefix="gpdb-${GPDB_RELEASE_TAG}/" --format=zip -9 refs/tags/"${GPDB_RELEASE_TAG}"
+	git archive --prefix "${PACKAGE}-${GPDB_RELEASE_TAG}/" --format "tar" --output "${ARCHIVE_TMP_DIR}/gpdb_src.tar" HEAD
+	git submodule foreach --recursive "echo \$displaypath: \$sha1: \$toplevel;
+	git archive --prefix=${PACKAGE}-${GPDB_RELEASE_TAG}/\$displaypath/ \
+		--format tar \$sha1 \
+		--output $ARCHIVE_TMP_DIR/submodule-\$sha1.tar"
```
Comment (amilkh) [inline ci/concourse/scripts/gpdb_github_release.bash]: suggest to indent these lines to make it clear they are part of `submodule foreach loop`

**Comment by baotingfang** [comment]
> overall looks good, but few points:
> 
> * let's add more details in commit message?
> * is it necessary to git archive tar, untar, and re-tar/zip? maybe there is a way to do it with `git submodule update` and then tar/zip

That's a good question. With @xinzweb, we provide another solution for the archiving source code.

ref: https://github.com/greenplum-db/greenplum-database-release/tree/168285479-archive-gpdb-src-2

File: ci/concourse/scripts/gpdb_github_release.bash (hunk 2 above)
Comment (baotingfang) [reply ci/concourse/scripts/gpdb_github_release.bash]: Usually, I run the 'shfmt -d -w .' to reformat the bash scripts. :)
